core/controller/src/main/scala/whisk/core/loadBalancer/InvokerSupervision.scala

```diff
@@ -105,10 +161,42 @@ class InvokerPool(kv: KeyValueStore, invokerDownCallback: String => Unit) extend
                 case _       =>
             }
     }
+
+    /** Receive activeAcks from Invoker. */
+    activeAckConsumer.onMessage((topic, _, _, bytes) => {
+        val raw = new String(bytes, "utf-8")
+        CompletionMessage.parse(raw) match {
+            case Success(m: CompletionMessage) => {
+                self ! InvocationFinishedMessage(m.invoker, m.response.response.statusCode <= 2)
+                activeAckCallback(m)
+            }
+            case Failure(t) => logging.error(this, s"failed processing message: $raw with $t")
+        }
+    })
+
+    /** Receive Ping messages from invokers. */
+    pingConsumer.onMessage((topic, _, _, bytes) => {
+        val raw = new String(bytes, "utf-8")
+        PingMessage.parse(raw) match {
+            case Success(p: PingMessage) => self ! p
+            case Failure(t)              => logging.error(this, s"failed processing message: $raw with $t")
+        }
+    })
 }
 
 object InvokerPool {
-    def props(kv: KeyValueStore, cb: String => Unit) = Props(new InvokerPool(kv, cb))
+    def props(kv: KeyValueStore,
+              cb: String => Unit, mp: MessageProducer,
+              ackC: MessageConsumer,
+              acb: CompletionMessage => Unit,
+              pc: MessageConsumer) = Props(new InvokerPool(kv, cb, mp, ackC, acb, pc))
+
+    val testAction = ExecManifest.runtimesManifest.resolveDefaultRuntime("nodejs").map { manifest =>
+        new WhiskAction(
+            namespace = EntityPath("whisk.system"),
+            name = EntityName("invokerHealthTestAction"),
+            exec = new CodeExecAsString(manifest, """function main(params) { return params || {}; }""", None))
+    }
```

Comment:
I moved it here now, but I left it as variable.
The reason is as discussed: Don't start if it is not possible to create the action.